### ISSUE TITLE
`partial` does not play nicely with `Unpack`

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#8617]
**Describe the bug**
A type incompatibility error is reported when attempting to fix arguments of a function with `Unpack`-annotated kwargs using `functools.partial` despite the types being correct.

**Code or Screenshots**
```py
# pyright: strict

from functools import partial
from typing import TypedDict, Unpack


class Kwargs(TypedDict, total=False):
    bar: str


def foo(**kwargs: Unpack[Kwargs]) -> None: ...


foo(bar='baz')  # Ok
make_foo = partial(foo, bar='baz')  # Error: Argument of type "Literal['baz']" cannot be assigned to parameter "bar" of type "*Kwargs" in function "foo"

```

**VS Code extension or command-line**
Either, 1.1.374

**Comment by erictraut** [comment]
This is included in pyright 1.1.375.
